### PR TITLE
Allow N/A instead of fallback to 1/1/1970 for empty dates.

### PR DESCRIPTION
**Pull request by lucaboesch**

### CODE REVIEW

**Comment by mattporritt** [comment]
hi @lucaboesch 
Thanks for doing these! I'll get the team to have a closer look but they look good to me.
It'd be great to hear how your team is using this plugin (feel free to reach out to me directly)

File: classes/quiz.php [hunk 1]
```diff
@@ -226,10 +226,26 @@ public function get_quiz_data(int $quizid): \stdClass {
         $overrideinfo = $this->get_quiz_override_info($quizid, $context);
         $questions = $this->get_quiz_questions($quizid);
         $frequency = new frequency();
-        $timesopen = userdate($quizrecord->timeopen, get_string('strftimedatetime', 'langconfig'));
-        $timeclose = userdate($quizrecord->timeclose, get_string('strftimedatetime', 'langconfig'));
-        $overrideinfostart = userdate($overrideinfo->start, get_string('strftimedatetime', 'langconfig'));
-        $overrideinfoend = userdate($overrideinfo->end, get_string('strftimedatetime', 'langconfig'));
+        if ((isset($quizrecord->timeopen) && ($quizrecord->timeopen != 0))) {
```
Comment (dmitriim) [inline classes/quiz.php:229]: is it the same as `if (!empty($quizrecord->timeopen)) {`

**Comment by dmitriim** [comment]
Hi @lucaboesch !

Thanks for working on the patch!

I have added one comment to the code. it's not super critical, but can make the code a little bit cleaner. 
More important: would be good to extend phpunit tests to cover those new date related scenarios. 
Also master branch has been updated recently, would be nice to rebase your changes on top of it.

File: classes/quiz.php (hunk 1 above)
Comment (lucaboesch) [reply classes/quiz.php]: Resolved, @dmitriim 
Thanks a lot,

Luca

Comment (dmitriim) [reply classes/quiz.php]: Thank you @lucaboesch ! I probably should be very specific. The same should apply to all extra if conditions as they are pretty much do the same thing, but with different variables.

**Comment by lucaboesch** [comment]
Oh, yes, of course, @dmitriim .
A was in the middle of something and not very present in thoughts.
Done!

Best,
Luca

**Comment by lucaboesch** [comment]
I'm closing this in favour of #126 which is the same idea but uses different branches

**Pull request closed without merging**